lib/chef/resource.rb

```diff
@@ -138,8 +138,11 @@ def initialize(name, run_context=nil)
     #
```

Comment:
Updated yard.  I think if we've documented that it returns a string and you subclass and return something else, you've clearly broken the API and its your fault.  That should be implied by the fact that we're now documenting the contract.